summerofcode/2023.md

```diff
@@ -382,7 +382,30 @@ OCI Artifacts are a set of conventions that allows us to store assets other than
 
 ### Kubebuilder
 
+#### Leverage Phase 2 Plugins API to allow creation and usage of external plugins
+
+- Description: Kubebuilder introduced Phase 2 Plugins to make itself more extensible as a CLI and as a library. This enables the community to develop, use, maintain and release their own plugins independently. Phase 2 Plugins has been available for a few releases, and in order to further increase and help facilitate adoption of this functionality, we need to work on creating an extensive sample, documentation and tutorial on how to build and use external plugins. To be able to achieve this goal, we can use go, ansible, python or helm to create the sample external plugin that will be used as a central reference plugin that serves as an example of an external plugin that leverages Phase 2 Plugins.
+
+- Expected outcome:
+  - Finish the meta task: <https://github.com/kubernetes-sigs/kubebuilder/issues/2600>
+    - Ensure the quality of the sample and Phase 2 Plugins API by:
+      - Adding e2e tests which should validate that the sample can be built and used with the Kubebuilder CLI.
+      - Implementing a maintainable solution to ensure that the sample will not get outdated.
+        - We might need to so something like: <https://github.com/kubernetes-sigs/kubebuilder/pull/3191>
+        - However, a dependabot for the sample might suffice in this case, just to ensure the dependencies.
+    - Review and improve the Plugin documentation to ensure that we the community can easily leverage the Phase 2 Plugins API: <https://book.kubebuilder.io/plugins/plugins.html>
+    - Verify that we are generating go docs for the Phase 2 Plugins API.
+    - Create a tutorial with the sample to let the community and users know how to build and use their own external plugins.
+    - Add an extra sample using any other language other than Golang.
+    - Create a video demo to showcase the Phase 2 Plugin working with the sample. This will demonstrate the achievements and will be shared with the community. We also will add the link to the demo in the docs.
+- Recommended Skills: Go
```

Comment:
```suggestion
- Recommended Skills: Go and one of Ansible, Python or Helm
```

I saw references to  go, ansible, python or helm above. You might want to list them here, but it is ok to leave them out as well if knowledge of these are not critical.

Reply:
I think only Go is required @rashmigottipati @everettraven WDYT?

Reply:
As a required skill, I think Go is good enough. Other skills are optional in my opinion.